src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -1174,6 +1175,63 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
         this->snapshots.back().udq.update(std::move(new_udq));
     }
 
+    void Schedule::handleUDT(HandlerContext& handlerContext)
+    {
+        auto new_udq = this->snapshots.back().udq();
+
+        using PUDT = ParserKeywords::UDT;
+
+        const auto& header = handlerContext.keyword.getRecord(0);
+        const std::string name = header.getItem<PUDT::TABLE_NAME>().get<std::string>(0);
+
+        const int dim = header.getItem<PUDT::DIMENSIONS>().get<int>(0);
+        if (dim != 1) {
+            throw OpmInputError("Only 1D UDTs are supported",
+                                handlerContext.keyword.location());
+
+        }
+
+        const auto& points = handlerContext.keyword.getRecord(1);
+        const std::string interp_type = points.getItem<PUDT::INTERPOLATION_TYPE>().get<std::string>(0);
+        UDT::InterpolationType type;
+        if (interp_type == "NV") {
+            type = UDT::InterpolationType::NearestNeighbour;
+        } else if (interp_type == "LC") {
+            type = UDT::InterpolationType::LinearClamp;
+        } else if (interp_type == "LL") {
+            type = UDT::InterpolationType::LinearExtrapolate;
+        } else {
+            throw OpmInputError(fmt::format("Unknown UDT interpolation type {}", interp_type),
+                                handlerContext.keyword.location());
+        }
+        const auto x_vals = points.getItem<ParserKeywords::UDT::INTERPOLATION_POINTS>().getData<double>();
+
+        if (!std::is_sorted(x_vals.begin(), x_vals.end())) {
+            throw OpmInputError("UDT: Interpolation points should be given in ascending order",
```

Comment:
"should" seems to weak. Maybe use "need to"?

Reply:
changed to need to